tests/controllers/handlers.spec.ts

```diff
@@ -0,0 +1,301 @@
+            const handler = createLegacyTilesRequestHandler({ 
+              map: mapComponentMock, 
+              features: featuresComponentMock,
+              logs: loggerComponentMock
+            })
+            const result = await handler({ url: new URL('http://localhost') }) as HandlerResponse
+            expect(result.status).toBe(301)
+            expect(result.headers?.location).toBe(`${s3Url}/v1`)
+          })
+        })
+
+        describe('and S3 URL is not available', () => {
+          beforeEach(() => {
+            mapComponentMock.getLastUploadedTilesUrl = jest.fn().mockReturnValue({})
+          })
+
+          it('should return 503 status since map is not ready and no S3 fallback', async () => {
+            const handler = createLegacyTilesRequestHandler({ 
+              map: mapComponentMock, 
+              features: featuresComponentMock,
+              logs: loggerComponentMock
+            })
+            const result = await handler({ url: new URL('http://localhost') }) as HandlerResponse
+            expect(result.status).toBe(503)
+            expect(result.body).toBe('Not ready')
+            expect(loggerWarnMock).toHaveBeenCalledWith('No S3 file available')
+          })
+        })
+      })
+    })
+
+    describe('when map is ready', () => {
+      beforeEach(() => {
+        mapComponentMock.isReady = jest.fn().mockReturnValue(true)
+      })
+
+      describe('and S3 redirect feature is ON', () => {
+        beforeEach(() => {
+          featuresComponentMock.getIsFeatureEnabled = jest.fn().mockResolvedValue(true)
+        })
+
+        describe('and S3 URL is available', () => {
+          it('should return 301 redirect to S3', async () => {
+            const handler = createLegacyTilesRequestHandler({ 
+              map: mapComponentMock, 
+              features: featuresComponentMock,
+              logs: loggerComponentMock
+            })
+            const result = await handler({ url: new URL('http://localhost') }) as HandlerResponse
+            expect(result.status).toBe(301)
+            expect(result.headers?.location).toBe(`${s3Url}/v1`)
+          })
+        })
+
+        describe('and S3 URL is not available', () => {
+          beforeEach(() => {
+            mapComponentMock.getLastUploadedTilesUrl = jest.fn().mockReturnValue({})
+          })
+
+          it('should serve tiles from map and log warning', async () => {
+            const handler = createLegacyTilesRequestHandler({ 
+              map: mapComponentMock, 
+              features: featuresComponentMock,
+              logs: loggerComponentMock
+            })
+            const result = await handler({ url: new URL('http://localhost') }) as HandlerResponse
+            expect(result.status).toBe(200)
+            expect(loggerWarnMock).toHaveBeenCalledWith('No S3 file available')
+            expect(result.headers?.['content-type']).toBe('application/json')
```

Comment:
This line, which is used in all tests that assert that the tiles are returned is not making sure the tiles are being returned as expected, as it only checks that the response is of type `application/json`. Would you mind changing the assertion to do so?